src/app/ngrx/reducers/episode.reducer.ts

```diff
@@ -1,78 +1,91 @@
-import { ActionTypes, CmsPodcastEpisodePageSuccessAction, CmsPodcastEpisodePageFailureAction, AllActions  } from '../actions';
-import { HalDoc } from 'ngx-prx-styleguide';
+import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
+import { Episode } from './models/episode.model';
+import { ActionTypes, AllActions } from '../actions';
 
-export const EPISODE_PAGE_SIZE = 10;
-
-export interface EpisodeModel {
-  doc?: HalDoc;
-  id: number;
-  seriesId: number;
-  title: string;
-  publishedAt: Date;
-  page?: number;
-  color?: string;
-  feederUrl?: string;
-  guid?: string;
+export interface State extends EntityState<Episode> {
+  // additional entities state properties
+  pagesLoaded: number[];
+  pagesLoading: number[];
+  total: number;
+  error?: any;
 }
 
-export interface EpisodeState {
-  entities?: {[id: number]: EpisodeModel};
```

Comment:
this _used_ to be the CMS episode reducer and was using the entity pattern manually for state shape. @ngrx/entity cleans that up a little bit, also you can use the @ngrx/schematics generator with @ngrx/entity to do the work for you, win/win.